app/services/platform-apps/api/modules/display.ts

```diff
@@ -67,40 +50,71 @@ export class DisplayModule extends Module {
   resize(ctx: IApiContext, displayId: string, size: IVec2) {
     const entry = this.getDisplayEntry(displayId);
 
-    entry.size = size;
-    this.updateDisplay(entry);
+    entry.options.size = size;
+    this.updateDisplay(entry, ctx.pageTransform.getValue());
   }
 
   @apiMethod()
   move(ctx: IApiContext, displayId: string, position: IVec2) {
     const entry = this.getDisplayEntry(displayId);
 
-    entry.position = position;
-    this.updateDisplay(entry);
+    entry.options.position = position;
+    this.updateDisplay(entry, ctx.pageTransform.getValue());
   }
 
   @apiMethod()
   destroy(ctx: IApiContext, displayId: string) {
     this.destroyDisplay(displayId);
   }
 
-  private updateDisplay(displayEntry: IDisplayEntry) {
-    if (displayEntry.webviewVisible) {
-      displayEntry.display.resize(displayEntry.size.x, displayEntry.size.y);
-      displayEntry.display.move(
-        displayEntry.webviewPosition.x + displayEntry.position.x,
-        displayEntry.webviewPosition.y + displayEntry.position.y,
-      );
-    } else {
-      displayEntry.display.resize(0, 0);
+  private updateDisplay(displayEntry: IDisplayEntry, transform: IBrowserViewTransform) {
+    // The container isn't mounted, so get rid of any active displays and stop
+    if (!transform.mounted) {
+      if (displayEntry.display) {
+        displayEntry.display.destroy();
```

Comment:
Could we extract method on this one into `disposeDisplay` or something like that? We could reuse below.